cli/pcluster/commands.py

```diff
@@ -540,13 +540,12 @@ def _persist_cloudwatch_log_groups(cluster_name):
     """Enable cluster's CloudWatch log groups to persist past cluster deletion."""
     LOGGER.info("Configuring {0}'s CloudWatch log groups to persist past cluster deletion.".format(cluster_name))
     substacks = utils.get_cluster_substacks(cluster_name)
-    substack_template_pairs = [(stack, utils.get_stack_template(stack.get("StackName"))) for stack in substacks]
-    substack_template_keys_triplets = [
-        (s, t, _get_unretained_cw_log_group_resource_keys(t)) for s, t in substack_template_pairs
-    ]
-    for stack, template, keys in substack_template_keys_triplets:
-        if keys:  # Only persist the CloudWatch group
-            _persist_stack_resources(stack, template, keys)
+    cw_substack = next((stack for stack in substacks if "CloudWatchLogsSubstack" in stack.get("StackName")), None)
+    if cw_substack:
+        cw_substack_template = utils.get_stack_template(cw_substack.get("StackName"))
+        log_group_keys = _get_unretained_cw_log_group_resource_keys(cw_substack_template)
+        if log_group_keys:  # Only persist the CloudWatch group
+            _persist_stack_resources(cw_substack, cw_substack_template, log_group_keys)
```

Comment:
Is there anything that might cause creation of the CloudWatch substack to fail, causing the update to fail for the same reason that's motivating this change?

Looking at the [CloudWatch quotas](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html) this seems unlikely, as the only thing on that page that seems like it would prevent  the creation of log group would be hitting the limit for number of log groups in a region, and 20,000 is a pretty large number for that.

Reply:
It's very unlikely but it could still happen. I slightly modified the error message in case of failure so that the user knows that he has to retry without keep-logs if he wants to delete anyway